CMakeLists.txt

```diff
@@ -80,9 +81,19 @@ string(APPEND CMAKE_CXX_FLAGS " -Wbuiltin-macro-redefined")
 string(APPEND CMAKE_CXX_FLAGS " -pedantic")
 string(APPEND CMAKE_CXX_FLAGS " -Werror")
 string(APPEND CMAKE_CXX_FLAGS " -fno-omit-frame-pointer")
-if(LEAKCHECK AND THREADCHECK)
-    message(FATAL_ERROR "Cannot have both Thread and Address/Leak Sanitizers Enabled")
+
+# At most, one sanitizer should be defined from below ALL_SANITIZER_OPTS
+set(COUNTER 0)
+set(ALL_SANITIZER_OPTS LEAKCHECK;THREADCHECK;UNDEFINED_BEHAVIOR_CHECK)
+foreach(option IN LISTS ALL_SANITIZER_OPTS)
+  if(${option})
+    MATH(EXPR COUNTER "${COUNTER}+1")
+  endif()
+endforeach()
+if(${COUNTER} GREATER 1)
+  message(FATAL_ERROR "More than one sanitizer was chosen!")
```

Comment:
Please give an error message which shows the chosen options in addition to their amount.